Snowplow/Internal/Session/SPSessionControllerImpl.m

```diff
@@ -117,7 +131,7 @@ - (NSString *)sessionId {
         SPLogTrack(nil, @"Attempt to access SessionController fields when disabled");
         return nil;
     }
-    return self.session.getSessionId;
+    return self.session.state.sessionId;
```

Comment:
This is a really tiny nitpick, but I find this change making it a bit worse – now the `SPSessionControllerImpl` needs to know more about the inner implementation of `session` to go through the `state` to access `sessionId`. This violates the Law of Demeter. But no need to change now.

Reply:
I wouldn't see this by this point of view. Session.state is not an inner implementation. State is exposed as a readonly property and exposed in the callback directly to the user. Also `getSessionId` has been removed. IMO Session should do its job of managing the session and expose a minimum interface. Exposing any single field of the session context was a good design until the Session class was a published class. State shouldn't be seen as the state of the Session manager but the state of the session from a user point of view. I think I had to do even more, removing `getSessionIndex` as the current implementation is a bit incoherent, as you noticed.